nebula-logger/plugins/log-retention-rules/plugin/classes/LogRetentionFilter.cls

```diff
@@ -0,0 +1,464 @@
+                when 'LESS_THAN_OR_EQUAL_TO' {
+                    return recordFieldValue <= comparisonValue;
+                }
+                when 'GREATER_THAN' {
+                    return recordFieldValue > comparisonValue;
+                }
+                when 'GREATER_THAN_OR_EQUAL_TO' {
+                    return recordFieldValue >= comparisonValue;
+                }
+                when else {
+                    throw new IllegalArgumentException('Unsupported operator for Datetime: ' + this.filterCondition.Operator__c);
+                }
+            }
+        }
+
+        private Boolean compareDecimal(Decimal recordFieldValue, Decimal comparisonValue) {
+            switch on this.filterCondition.Operator__c {
+                when 'EQUAL_TO' {
+                    return recordFieldValue == comparisonValue;
+                }
+                when 'NOT_EQUAL_TO' {
+                    return recordFieldValue != comparisonValue;
+                }
+                when 'LESS_THAN' {
+                    return recordFieldValue < comparisonValue;
+                }
+                when 'LESS_THAN_OR_EQUAL_TO' {
+                    return recordFieldValue <= comparisonValue;
+                }
+                when 'GREATER_THAN' {
+                    return recordFieldValue > comparisonValue;
+                }
+                when 'GREATER_THAN_OR_EQUAL_TO' {
+                    return recordFieldValue >= comparisonValue;
+                }
+                when else {
+                    throw new IllegalArgumentException('Unsupported operator for Decimal: ' + this.filterCondition.Operator__c);
+                }
+            }
+        }
+
+        private Boolean compareString(String recordFieldValue, String comparisonValue) {
+            switch on this.filterCondition.Operator__c {
+                when 'EQUAL_TO' {
+                    return recordFieldValue == comparisonValue;
+                }
+                when 'NOT_EQUAL_TO' {
+                    return recordFieldValue != comparisonValue;
+                }
+                when 'STARTS_WITH' {
+                    return recordFieldValue.startsWith(comparisonValue);
+                }
+                when 'CONTAINS' {
+                    return recordFieldValue.contains(comparisonValue);
+                }
+                when 'ENDS_WITH' {
+                    return recordFieldValue.endsWith(comparisonValue);
+                }
+                when else {
+                    throw new IllegalArgumentException('Unsupported operator for String: ' + this.filterCondition.Operator__c);
+                }
+            }
+        }
+    }
+
+    // Credit goes to this StackExchange post for the original BooleanExpression class -
+    // below is a modified version of the class
+    // https://salesforce.stackexchange.com/questions/113300/boolean-evaluation-in-apex/113308
+    @SuppressWarnings('PMD.ApexDoc')
+    private class BooleanExpression {
```

Comment:
happy to see this thing resurrected - it's truly wild!

Reply:
It's absolutely nuts!